src/prefect/server/schemas/actions.py

```diff
@@ -235,6 +236,19 @@ def _validate_parameters_conform_to_schema(cls, value, values):
     def _validate_parameter_openapi_schema(cls, value, values):
         return validate_parameter_openapi_schema(value, values)
 
+    @validator("parameters")
+    def validate_parameters_for_json_compliance_and_size(cls, parameters: dict):
+        try:
+            serialized_parameters = orjson.dumps(parameters)
+        except Exception as e:
+            raise ValueError("Flow run parameters must be serializable to JSON.") from e
+
+        if len(serialized_parameters) > 512_000:
```

Comment:
do you think this should be a setting?

its possible (albeit unlikely) that those running open source servers could currently be sending parameters larger than this and so this would be a breaking change for them